public/app/features/explore/QueryEditor.tsx

```diff
@@ -45,13 +45,13 @@ export default class QueryEditor extends PureComponent<QueryEditorProps, any> {
         target,
         refresh: () => {
           setTimeout(() => {
-            this.props.onQueryChange(target);
-            this.props.onExecuteQuery();
+            this.props.onQueryChange?.(target);
```

Comment:
This looks a bit weird. How come this did not crash before? Seems like the code assumes these functions to be non optional so maybe we should change the props to be non optional?

Reply:
Not sure, but [QueryEditorRow](https://github.com/grafana/grafana/blob/b756aa0bb197286963640ebb29501c82c04f3725/public/app/features/dashboard/panel_editor/QueryEditorRow.tsx#L153) doesn't seem to pass these props for example, so I'm not sure we can just make them non-optional.

Reply:
Ah ok in that case I assume the code before could crash in some cases.